docs/content/en/docs/yaml-crd-ref/taskdefinition.md

```diff
@@ -3,3 +3,310 @@ title: KeptnTaskDefinition
+      that is used in multiple places, possibly with different parameters.
+      An example is:
+
+       ```yaml
+       spec:
+         function:
+           functionRef:
+             name: slack-notification
+       ```
+
+      This can also be used to group a set of tasks
+      into a single `KeptnTaskDefinition`,
+      such as defining a `KeptnTaskDefinition` for testing.
+      In this case, it other, existing `KeptnTaskDefinition`s
+      for each type of test to be run,
+      specifying each by the value of the `name` field.
+
+  * **parameters** - An optional field to supply input parameters to a function.
+    The Lifecycle Toolkit passes the values defined inside the `map` field
+    as a JSON object.
+    For example:
+
+     ```yaml
+       spec:
+         parameters:
+           map:
+             textMessage: "This is my configuration"
+     ```
+
+     The JSON object can be read
+     through the `DATA` environment variable using `Deno.env.get("DATA");`.
+
+     Multi-level maps are not supported at this time.
+
+     Currently only one secret can be passed.
+     The secret must have a `key` called `SECURE_DATA`.
+     It can be accessed via the environment variable `Deno.env.get("SECURE_DATA")`.
+     See [Context](#context) for details.
+
+  * **secureParameters** -- An optional field used to pass a Kubernetes secret.
+    The `secret` value is the Kubernetes secret name
+    that is mounted into the runtime and made available to functions
+    using the `SECURE_DATA` environment variable.
+    For example:
+
+    ```yaml
+    secureParameters:
+      secret: slack-token
+    ```
+
+    See [Create secret text](#create-secret-text) for details.
+
+## Usage
+
+A Task is responsible for executing the TaskDefinition of a
+[workload](https://kubernetes.io/docs/concepts/workloads/)..
+The execution is done by spawning a Kubernetes
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/)
+to handle a single Task.
+In its state, it tracks the current status of this Kubernetes Job.
+
+### Context
+
+A Kubernetes context is a set of access parameters
+that contains a Kubernetes cluster, a user, and a namespace.
+For more information, see
+[Configure Access to Multiple Clusters](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/).
+
+A context environment variable is available via `Deno.env.get("CONTEXT")`.
+It can be used like this:
+
+```javascript
+let context = Deno.env.get("CONTEXT");
+
+if (contextdata.objectType == "Application") {
+    let application_name = contextdata.appName;
+    let application_version = contextdata.appVersion;
+}
+
+if (contextdata.objectType == "Workload") {
+    let application_name = contextdata.appName;
+    let workload_name = contextdata.workloadName;
+    let workload_version = contextdata.workloadVersion;
+}
+```
+
+### Create secret text
+
+```yaml
+# kubectl create secret generic my-secret --from-literal=SECURE_DATA=foo
```

Comment:
I find this a little bit misleading, as it seems that it's part of yaml. We can maybe add a paragraph like:

To create a secret used in the KeptnTask execute this command:
kubectl create secret generic my-secret --from-literal=SECURE_DATA=foo

Or something in this manner.

Users might find it confusing seeing a kubectl command together with a yaml manifest

Reply:
Good stuff!  I added your lead-in to the command and also moved this section to the guide chapter, along with the info about context.